azurerm/resource_arm_mysql_configuration.go

```diff
@@ -62,19 +75,19 @@ func resourceArmMySQLConfigurationCreate(d *schema.ResourceData, meta interface{
 
 	future, err := client.CreateOrUpdate(ctx, resourceGroup, serverName, name, properties)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error issuing create/update request for MySQL Configuration %s (resource group %s, server name %s): %v", name, resourceGroup, serverName, err)
 	}
 
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return err
+		return fmt.Errorf("Error waiting pm create/update future for MySQL Configuration %s (resource group %s, server name %s): %v", name, resourceGroup, serverName, err)
```

Comment:
I'd argue users don't need to know about the future (and that instead the error message is more helpful as "waiting for create/update of {x}") - since that's an implementation detail?

```suggestion
		return fmt.Errorf("Error waiting for create/update of MySQL Configuration %s (resource group %s, server name %s): %v", name, resourceGroup, serverName, err)
```